doc/source/user/numpy-for-matlab-users.rst

```diff
@@ -25,165 +25,50 @@ Some Key Differences
-       operations.  One needs to use specific functions for linear algebra
-       (though for matrix multiplication, one can use the ``@`` operator
-       in python 3.5 and above).
-
-   * - MATLAB® uses 1 (one) based indexing. The initial element of a
+   * - In MATLAB, the basic type is a multidimensional array. By default, MATLAB
+       creates two-dimensional arrays of double precision floating point numbers.
+       Operations on the 2D instances of these arrays are modeled on
+       matrix operations in linear algebra. 
+     - In NumPy, the basic type is a multidimensional ``array``. By default,
+       NumPy creates 0-D or 1-D arrays for inputs of numbers or lists,
+       respectively.  Operations on these arrays in all dimensionalities
+       including 2D are element-wise operations.  One needs to use specific
+       functions for linear algebra (though for matrix multiplication, one can
+       use the ``@`` operator in Python 3.5 and above).
+
+   * - MATLAB uses 1 (one) based indexing. The initial element of a
        sequence is found using a(1).
        :ref:`See note INDEXING <numpy-for-matlab-users.notes>`
      - Python uses 0 (zero) based indexing. The initial element of a
        sequence is found using a[0].
 
-   * - MATLAB®'s scripting language was created for doing linear algebra.
-       The syntax for basic matrix operations is nice and clean, but the API
-       for adding GUIs and making full-fledged applications is more or less
-       an afterthought.
-     - NumPy is  based on Python, which was designed from the outset to be
-       an excellent general-purpose programming language.  While Matlab's
+   * - MATLAB's scripting language was created for doing linear algebra so the
        syntax for some array manipulations is more compact than
-       NumPy's, NumPy (by virtue of being an add-on to Python) can do many
-       things that Matlab just cannot, for instance dealing properly with
-       stacks of matrices.
-
-   * - In MATLAB®, arrays have pass-by-value semantics, with a lazy
-       copy-on-write scheme to prevent actually creating copies until they
+       NumPy's. The syntax for basic matrix operations is nice and clean, but the
+       API for adding GUIs and making full-fledged applications is more or less
+       an afterthought.
+     - NumPy is  based on Python, which was designed from the outset to be an
+       excellent general-purpose programming language.  The advantage to NumPy
+       is access to Python libraries including: `SciPy
+       <https://www.scipy.org/>`_, `Matplotlib <https://matplotlib.org/>`_,
+       `Pandas <https://pandas.pydata.org/>`_, `OpenCV <https://opencv.org/>`_,
+       and more. In addition to Python-specific packages, much `scientific
+       computing software
+       <https://en.wikipedia.org/wiki/List_of_Python_software#Embedded_as_a_scripting_language>`_
+       use Python as a scripting language. 
```

Comment:
You don't have to word it exactly this way, but I thought since this was a guide to newcomers it would be helpful to give a short description of these packages. 

Note the lower-case spelling of matplotlib and pandas.

The Wikipedia entry seemed off-topic (it lists programs that can call out to Python rather than packages a Python program can call).


```suggestion
     - Because NumPy is based on Python, a general-purpose programming 
     language, users have access to the entire Python "scientific ecosystem," 
     whose packages include the computational tools of `SciPy
       <https://www.scipy.org/>`_, the plotting software of 
       `matplotlib <https://matplotlib.org/>`_,
      the data manipulation and analysis tool `pandas <https://pandas.pydata.org/>`_, 
      the image-processing library`OpenCV <https://opencv.org/>`_, and more.
```

Reply:
> The Wikipedia entry seemed off-topic (it lists programs that can call out to Python rather than packages a Python program can call).

Agreed. https://scipy.org/topical-software.html may be a better link.

Reply:
> > The Wikipedia entry seemed off-topic (it lists programs that can call out to Python rather than packages a Python program can call).
> 
> Agreed. https://scipy.org/topical-software.html may be a better link.

The scipy.org is a good link to see where Python is used to build software. 

> The Wikipedia entry seemed off-topic (it lists programs that can call out to Python rather than packages a Python program can call).

I like the wiki entry because it lists scientific computing software that use Python as a scripting language. Which means you can create NumPy-based scripts to run FEA/CFD/dynamics/etc. without having to build it yourself. Then, you can post-process the data. We can drop it if its not of interest to the broader MATLAB conversions, but its one of the benefits people coming from MATLAB might not consider at first.